File: Album/src/main/java/com/hqq/album/customize/SquareImageView.java
package com.hqq.album.customize;

import android.content.Context;
import androidx.annotation.Nullable;
import android.util.AttributeSet;

/**
 * @Author : huangqiqiang
 * @Package : com.rongji.shangwenwan.weight
 * @FileName :   SquareImageView
 * @Date : 2018/7/26 0026  下午 5:41
 * @Descrive : TODO
 * @Email :
 * 宽度为准的 正方形ImageView
 */

public class SquareImageView extends androidx.appcompat.widget.AppCompatImageView {
    public SquareImageView(Context context) {
        super(context);
    }

    public SquareImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec);
    }
}

File: Album/src/main/java/com/easy/album/customize/SquareImageView.kt
package com.easy.album.customize

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * @Author : huangqiqiang
 * @Package : com.rongji.shangwenwan.weight
 * @FileName :   SquareImageView
 * @Date : 2018/7/26 0026  下午 5:41
 * @Descrive : TODO
 * @Email :
 * 宽度为准的 正方形ImageView
 */
class SquareImageView : AppCompatImageView {
    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {}

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec)
    }
}